test/e2e/bootstrapper_test.go

```diff
@@ -62,7 +62,7 @@ func TestBootstrapper(t *testing.T) {
 	t.Run("Uninstall kubelet without kubelet service present", testUninstallWithoutKubeletSvc)
 
 	// Run the bootstrapper, which will start the kubelet service
-	wmcb, err := bootstrapper.NewWinNodeBootstrapper(installDir, ignitionFilePath, kubeletPath, "", "", "", "")
+	wmcb, err := bootstrapper.NewWinNodeBootstrapper(installDir, ignitionFilePath, kubeletPath, "", "", "", "", "")
```

Comment:
The provider should be discovered as part of the test and then passed in to all invocations of `NewWinNodeBootstrapper`, excluding the uninstall kubelet tests, as arguments are ignored there.

A test should be added to check that the `hostname-override` argument is set.

Do we need to check the value? What is the expected behavior if the value is incorrect?

Reply:
I put "azure" platform as the parameter value there at the moment, in this case wcmb doesn't override the hostname. In case of aws, it will send requests to AWS metadata service, which is hard to emulate in these tests.